StashSearch/StashComponent.cs

```diff
@@ -82,25 +82,58 @@ private void Start()
             Plugin.SearchControllers.Add(_searchController);
         }
 
+        private void OnEnable()
+        {
+            // skip enabling search bar if loaded into a map
+            if (Singleton<GameWorld>.Instantiated)
+            {
+                return;
+            }
+
+            // enable the search bar
+            _searchObject.SetActive(true);
+            _searchRestoreButtonObject.SetActive(true);
+
+            // move and resize the complex stash to accomidate the search bar
+            _oldComplexStashSizeDelta = _complexStash.RectTransform.sizeDelta;
+            _oldComplexStashLocalPosition = _complexStash.Transform.localPosition;
+            _complexStash.RectTransform.sizeDelta = new Vector2(680, -260);
+            _complexStash.Transform.localPosition = new Vector3(948, 12, 0);
+            _hasMovedComplexStash = true;
+        }
+
         private void OnDisable()
         {
+            // reset input textbox
             _inputField.text = string.Empty;
+
+            // disable the search bar
+            _searchObject.SetActive(false);
+            _searchRestoreButtonObject.SetActive(false);
+
+            // return _complexStash to normal position
+            if (_hasMovedComplexStash)
+            {
+                _complexStash.RectTransform.sizeDelta = _oldComplexStashSizeDelta;
+                _complexStash.Transform.localPosition = _oldComplexStashLocalPosition;
+                _hasMovedComplexStash = false;
+            }
+            
+            // NOTE: could potentially clear search here rather than having the OnScreenChangedPatch do it
         }
 
         private void Update()
         {
             if (StashSearchConfig.FocusSearch.Value.IsDown() && OnScreenChangedPatch.CurrentScreen == EEftScreenType.Inventory)
```

Comment:
I think that the check for `CurrentScreen` is redundant here, since `Update()` only runs if this script is active. Since it's parented to `InventoryScreen` ultimately, when that isn't active, this doesn't run.